src/Data/Array.purs

```diff
@@ -571,6 +842,11 @@ group' = group <<< sort
 
 -- | Group equal, consecutive elements of an array into arrays, using the
 -- | specified equivalence relation to detemine equality.
+-- |
+-- | ```purescript
+-- | groupBy (\a b → odd a && odd b) [1, 1, 2, 2, 3, 3] = [[1, 1], 2, 2, [3, 3]]
```

Comment:
For me this prints `[(NonEmpty 1 [1]),(NonEmpty 2 []),(NonEmpty 2 []),(NonEmpty 3 [3])]` which is not too pretty, but the 2s should be in singleton arrays at least: `[[1, 1], [2], [2], [3, 3]]`

Reply:
I used the array-notation, because the example for `group` also used that notation

Reply:
Ah, I see. @garyb looks like you made the switch to `NonEmpty` (in 738f1b4d35eac922f21e64ef01e9f5002df8e92d). Did you intentionally not change the examples for `group` and `group'`?

Reply:
No, it wasn't intentional :)

There is something nice about not showing the `NonEmpty` stuff in the example, as it does make it a bit clearer to see what is going on, but it's inaccurate obviously so I'm not sure which is better really. I guess we do show the exact output usually, so it probably should be in there. I don't know.